google/cloud/aiplatform/models.py

```diff
@@ -2044,32 +2064,44 @@ def upload(
                 "Both `explanation_metadata` and `explanation_parameters` should be specified or None."
             )
 
-        api_client = cls._instantiate_client(
-            location, credentials, appended_user_agent=appended_user_agent
-        )
-        env = None
-        ports = None
-
-        if serving_container_environment_variables:
-            env = [
-                gca_env_var_compat.EnvVar(name=str(key), value=str(value))
-                for key, value in serving_container_environment_variables.items()
-            ]
-        if serving_container_ports:
-            ports = [
-                gca_model_compat.Port(container_port=port)
-                for port in serving_container_ports
-            ]
+        appended_user_agent = None
+        if local_model:
+            container_spec = local_model.get_serving_container_spec()
+            if not container_spec.image_uri:
```

Comment:
container_spec.image_uri doesn't get used in this code. Let the code that does use container_spec.image_uri throw if it isn't present. Remove this check.

Reply:
Done.